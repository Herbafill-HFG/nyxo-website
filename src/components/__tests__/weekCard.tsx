import React from "react"
import renderer from "react-test-renderer"

import WeekCard from "../week/WeekCard"

describe("Week card", () => {
  const name = "Week name"
  const path = "/week/week-name"
  const key = path
  const duration = 15
  const intro = "Lorem ipsum"
  const lessons = [
    {
      lessonName: "Why We Sleep?",
      slug: "why-we-sleep",
    },
    {
      lessonName: "Internal Clock",
      slug: "internal-clock",
    },
    {
      lessonName: "Sleep Pressure",
      slug: "sleep-pressure",
    },
    {
      lessonName: "Is Napping Bad for You?",
      slug: "is-napping-bad-for-you",
    },
    {
      lessonName: "Do You Sleep Enough?",
      slug: "do-you-sleep-enough",
    },
    {
      lessonName: "Sleep Quality",
      slug: "sleep-quality",
    },
    {
      lessonName: "Sleep Hygiene",
      slug: "sleep-hygiene",
    },
  ]
  const coverPhoto = "cat.jpeg"

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <WeekCard
          key={key}
          name={name}
          path={path}
          duration={duration}
          intro={intro}
          lessons={lessons}
          coverPhoto={coverPhoto}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
