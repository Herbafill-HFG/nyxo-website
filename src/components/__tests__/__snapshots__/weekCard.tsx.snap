// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week card renders correctly 1`] = `
<a
  className="sc-fznZeY gCXwDP"
  href="/week/week-name"
>
  <div
    className="sc-fzokOt bJFgDa"
  >
    <div
      className="sc-fzqBZW ieGBzv gatsby-image-wrapper"
      style={
        Object {
          "maxHeight": null,
          "maxWidth": null,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt=""
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <button
      className="sc-fzozJi kVUjYJ"
      onClick={[Function]}
    >
      <svg
        className="sc-AxmLO huqxQS"
        fill="none"
        height="25px"
        stroke="#F42D97"
        strokeWidth="1.5"
        version="1.1"
        viewBox="0 0 28 30"
        width="25px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M23.3,5.076a6.582,6.582,0,0,0-10.446-1.71L12,4.147l-.827-.753A6.522,6.522,0,0,0,5.485,1.588,6.472,6.472,0,0,0,.7,5.075,6.4,6.4,0,0,0,1.91,12.544L11.283,22.2a1,1,0,0,0,1.434,0l9.36-9.638A6.414,6.414,0,0,0,23.3,5.076Z"
          />
        </g>
      </svg>
    </button>
    <div
      className="sc-fzoXzr dRZYpr"
    >
      <span
        className="sc-fzpjYC gZSBqH"
      >
        <svg
          className="sc-AxmLO huqxQS"
          fill="none"
          height="20px"
          stroke="currentColor"
          strokeWidth="1.5"
          version="1.1"
          viewBox="0 0 30 30"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M 2.25,1.5h19.5c0.414,0,0.75,0.336,0.75,0.75v1.5c0,0.414-0.336,0.75-0.75,0.75H2.25 C1.836,4.5,1.5,4.164,1.5,3.75v-1.5C1.5,1.836,1.836,1.5,2.25,1.5z "
            />
            <path
              d="M 3,4.5h18V18H3V4.5z "
            />
            <path
              d="M 12,21.5V18 "
            />
            <path
              d="M 1.5,18h21 "
            />
            <path
              d="M 15,9h3 "
            />
            <path
              d="M 15,12h3 "
            />
            <path
              d="M 9,7.5c1.657,0,3,1.343,3,3s-1.343,3-3,3s-3-1.343-3-3S7.343,7.5,9,7.5z "
            />
            <path
              d="M 13.5,22.5c0-0.553-0.672-1-1.5-1s-1.5,0.447-1.5,1"
            />
          </g>
        </svg>
        7
         Lessons
      </span>
    </div>
  </div>
  <div
    className="sc-fzqNJr havePp"
  >
    <h3
      className="sc-fzoyAV eXHTwC"
    >
      Week name
    </h3>
    <span
      className="sc-fzoLag dVakYK"
    >
      Lorem ipsum
    </span>
  </div>
</a>
`;
